factory/alchemy.py

```diff
@@ -90,6 +91,13 @@ def _get_or_create(cls, model_class, session, args, kwargs):
     @classmethod
     def _create(cls, model_class, *args, **kwargs):
         """Create an instance of the model, and save it to the database."""
+        session_factory = cls._meta.sqlalchemy_session_factory
+        if session_factory:
+            if cls._meta.sqlalchemy_session:
+                raise RuntimeError("Provide either a sqlalchemy_session or a sqlalchemy_session_factory, not both")
+
+            cls._meta.sqlalchemy_session = session_factory()
+
```

Comment:
That won’t work, because the modified metaclass will remain modified for the factory. The options should be checked at import time (probably using a [checker](https://github.com/FactoryBoy/factory_boy/blob/d2447abbbf4da99d9e76909b8657524cf8e1c48a/factory/base.py#L117)).

Example test modification leading to a failure:

```diff
diff --git a/tests/test_alchemy.py b/tests/test_alchemy.py
index 6a3ff62..98ac188 100644
--- a/tests/test_alchemy.py
+++ b/tests/test_alchemy.py
@@ -287,6 +287,7 @@ class SQLAlchemySessionFactoryTestCase(unittest.TestCase):
     def test_create_get_session_from_sqlalchemy_session_factory(self):
         SessionGetterFactory.create()
         self.assertEqual(SessionGetterFactory._meta.sqlalchemy_session, models.session)
+        SessionGetterFactory.create()
 
     def test_create_raise_exception_sqlalchemy_session_factory_not_callable(self):
         message = "Provide either a sqlalchemy_session or a sqlalchemy_session_factory, not both"
```

```
ERROR: test_create_get_session_from_sqlalchemy_session_factory (tests.test_alchemy.SQLAlchemySessionFactoryTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/freitafr/dev/factory_boy/tests/test_alchemy.py", line 290, in test_create_get_session_from_sqlalchemy_session_factory
    SessionGetterFactory.create()
  File "/home/freitafr/dev/factory_boy/factory/base.py", line 528, in create
    return cls._generate(enums.CREATE_STRATEGY, kwargs)
  File "/home/freitafr/dev/factory_boy/factory/alchemy.py", line 52, in _generate
    return super()._generate(strategy, params)
  File "/home/freitafr/dev/factory_boy/factory/base.py", line 465, in _generate
    return step.build()
  File "/home/freitafr/dev/factory_boy/factory/builder.py", line 264, in build
    instance = self.factory_meta.instantiate(
  File "/home/freitafr/dev/factory_boy/factory/base.py", line 317, in instantiate
    return self.factory._create(model, *args, **kwargs)
  File "/home/freitafr/dev/factory_boy/factory/alchemy.py", line 97, in _create
    raise RuntimeError("Provide either a sqlalchemy_session or a sqlalchemy_session_factory, not both")
RuntimeError: Provide either a sqlalchemy_session or a sqlalchemy_session_factory, not both
```